x-pack/plugins/observability/public/embeddable/slo/slo_selector.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useMemo, useState } from 'react';
+import { EuiComboBox, EuiComboBoxOptionOption, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { debounce } from 'lodash';
+import { ALL_VALUE, SLOWithSummaryResponse } from '@kbn/slo-schema';
+import { useFetchSloList } from '../../hooks/slo/use_fetch_slo_list';
+
+interface Props {
+  initialSlo?: SLOWithSummaryResponse;
+  onSelected: (slo: SLOWithSummaryResponse | undefined) => void;
+  hasError?: boolean;
+}
+
+const SLO_REQUIRED = i18n.translate('xpack.observability.sloEmbeddable.config.errors.sloRequired', {
+  defaultMessage: 'SLO is required.',
+});
+
+export function SloSelector({ initialSlo, onSelected, hasError }: Props) {
+  const [options, setOptions] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
+  const [selectedOptions, setSelectedOptions] = useState<Array<EuiComboBoxOptionOption<string>>>();
+  const [searchValue, setSearchValue] = useState<string>('');
+  const { isInitialLoading, isLoading, sloList } = useFetchSloList({ search: searchValue });
```

Comment:
I filter by the search term. It highlights the matched string, but it doesn't filter out the not matched strings. Need to check it.

Reply:
We need to use the `kqlQuery` instead of `search` (does not exist) and build the kqlQuery based on the `slo.name` field, e.g. `slo.name : searchValue`

Reply:
Hm, let's have a look at this one together. I couldn't find a quick way to make this work. Displayed name currenly contains the name, the groupBy and instanceId as well, so `slo.name : searchValue` wouldn't work.